storm-core/src/jvm/org/apache/storm/callback/Callback.java

```diff
@@ -18,6 +18,9 @@
 
 package org.apache.storm.callback;
 
+import clojure.lang.IFn;
+
+// To remove IFn after porting all callbacks to java
 public interface Callback {
     public <T> Object execute(T... args);
```

Comment:
I don't like this interface.  The generic <T> for the parameters is just confusing because it implies that all of the parameters should be of the same type, when in reality they are not.  I would much rather see something like

```
public interface ZKStateChangedCallback {
    public void changed(Watcher.Event.EventType type, String path);
}
```